iroh-base/src/key.rs

```diff
@@ -162,8 +110,9 @@ impl PublicKey {
     /// a valid `ed25519_dalek` curve point. Will never fail for bytes return from [`Self::as_bytes`].
     /// See [`VerifyingKey::from_bytes`] for details.
     pub fn from_bytes(bytes: &[u8; 32]) -> Result<Self, SignatureError> {
-        get_or_create_crypto_keys(bytes, |item| item.verifying_key)?;
-        Ok(Self(*bytes))
+        let key = VerifyingKey::from_bytes(bytes)?;
```

Comment:
Does it still make sense for the error to be VerifyingKey::SignatureError? That way we have a _public_ dependency on ed25519_dalek.

If we have this public dependency anyway, it would be nice if we had a way to get at the inner `CompressedEdwardsY` - in general provide simple conversions from our primitives to ed25519_dalek. If not, we could return our own error here.

Reply:
Ah, never mind. It seems we already have conversions. We didn't last time I checked this, which was some time ago.